sklearn/cluster/_hdbscan/_tree.pyx

```diff
@@ -282,40 +282,39 @@ cdef max_lambdas(cnp.ndarray hierarchy):
     return deaths
 
 
-cdef class TreeUnionFind (object):
+@final
+cdef class TreeUnionFind:
 
-    cdef cnp.ndarray _data_arr
-    cdef cnp.intp_t[:, ::1] _data
-    cdef cnp.ndarray is_component
+    cdef cnp.intp_t[:, ::1] data
+    cdef cnp.uint8_t[::1] is_component
 
     def __init__(self, size):
-        self._data_arr = np.zeros((size, 2), dtype=np.intp)
-        self._data_arr.T[0] = np.arange(size)
-        self._data = self._data_arr
-        self.is_component = np.ones(size, dtype=bool)
-
-    cdef union_(self, cnp.intp_t x, cnp.intp_t y):
+        cdef cnp.ndarray[cnp.intp_t, ndim=2] data_arr
+        data_arr = np.zeros((size, 2), dtype=np.intp)
+        data_arr.T[0] = np.arange(size)
```

Comment:
Seems a little waste to create a new array just for setting the values of another array. What is the performance like for doing it manually for large `size`?

```python
for i in range(size):
    data_arr[i, 0] = i
```